libdrgn/kdump.c

```diff
@@ -138,6 +204,30 @@ struct drgn_error *drgn_program_set_kdump(struct drgn_program *prog)
 #endif
 		if (err)
 			goto err;
+	} else {
+		char *vmcoreinfo = memdup(prog->vmcoreinfo.raw, prog->vmcoreinfo.raw_size);
+		if (!vmcoreinfo) {
+			err = &drgn_enomem;
+			goto err;
+		}
+		kdump_blob_t *blob = kdump_blob_new(vmcoreinfo, prog->vmcoreinfo.raw_size);
+		if (!blob) {
+			free(vmcoreinfo);
+			err = &drgn_enomem;
+			goto err;
+		}
+		kdump_attr_t attr;
+		attr.type = KDUMP_BLOB;
+		attr.val.blob = blob;
+		ks = kdump_set_attr(ctx, "linux.vmcoreinfo.raw", &attr);
+		if (ks != KDUMP_OK) {
+			free(vmcoreinfo);
```

Comment:
From my understanding of the libkdumpfile documentation, `blob` owns `vmcoreinfo` now, so this would cause `kdump_block_decref()` to double free, right?
```suggestion
```

Reply:
Yeah, you're definitely correct, and it really obviously looks  wrong too. Good catch.

More interestingly however is that the libkdumpfile code seems to indicate that even in the error case, the reference to the blob is already stolen, and thus we cannot free it in our error path. See [this comment](https://github.com/ptesarik/libkdumpfile/blob/tip/src/kdumpfile/attr.c#L898-L903). However, there are clearly cases of errors inside of `kdump_set_attr()` where the function returns and the blob is not decref'd. In summary: on some failures, we can't even decref the blob. On other failures, we should. And there's no way to tell 😜 

So my vote here (I know it's gross) is to heavily comment this and get rid of both the `free(vmcoreinfo)` and the `kdump_blob_decref()`.

Reply:
Ok, I filed https://github.com/ptesarik/libkdumpfile/issues/81 for Petr, we'll see if he agrees too. Assuming he does, I'll remove the error handling and document the leak .

Reply:
If `kdump_set_attr()` is supposed to take ownership, then yes, we should definitely remove `kdump_blob_decref()`, and I don't think we even need a comment; we just need to report any leaks as a libkdumpfile bug.

Reply:
Ok, dropped the comment and the `kdump_blob_decref()`, we can revisit it if Petr says the ownership semantic is supposed to be different.